File: frontend/lib/data/models/journey.dart
class JourneyDetails {
  final int journeyId;
  final String startTime;
  final String endTime;
  final int trainId;
  final int startStationId;
  final int endStationId;
  final String startStationName;
  final String endStationName;

  JourneyDetails({
    required this.journeyId,
    required this.startTime,
    required this.endTime,
    required this.trainId,
    required this.startStationId,
    required this.endStationId,
    required this.startStationName,
    required this.endStationName,
  });

  factory JourneyDetails.fromJson(Map<String, dynamic> json) {
    return JourneyDetails(
      journeyId: json['journey_id'] as int,
      startTime: json['start_time'] as String,
      endTime: json['end_time'] as String,
      trainId: json['train_id'] as int,
      startStationId: json['start_station_id'] as int,
      endStationId: json['end_station_id'] as int,
      startStationName: json['start_station_name'] as String,
      endStationName: json['end_station_name'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'journey_id': journeyId,
      'start_time': startTime,
      'end_time': endTime,
      'train_id': trainId,
      'start_station_id': startStationId,
      'end_station_id': endStationId,
      'start_station_name': startStationName,
      'end_station_name': endStationName,
    };
  }
}

class CreateJourneyRequest {
  final String startTime;
  final String endTime;
  final int trainId;
  final int startStationId;
  final int endStationId;

  CreateJourneyRequest({
    required this.startTime,
    required this.endTime,
    required this.trainId,
    required this.startStationId,
    required this.endStationId,
  });

  Map<String, dynamic> toJson() {
    return {
      'start_time': startTime,
      'end_time': endTime,
      'train_id': trainId,
      'start_station_id': startStationId,
      'end_station_id': endStationId,
    };
  }
}

File: backend/src/models/journey.rs
use serde::{Serialize, Deserialize};
use chrono::{DateTime, NaiveDateTime, Utc};

#[derive(Debug, Serialize, sqlx::FromRow)]
pub struct JourneyDetailedResponse {
    pub journey_id: i64,
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub train_id: Option<i64>,
    pub start_station_id: Option<i64>,
    pub end_station_id: Option<i64>,
    pub start_station_name: Option<String>,
    pub end_station_name: Option<String>,
}

#[derive(Debug, Serialize, sqlx::FromRow)]
pub struct JourneyResponse {
    pub journey_id: i64,
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub train_id: Option<i64>,
    pub start_station_id: Option<i64>,
    pub end_station_id: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct CreateJourney {
    pub start_time: String,
    pub end_time: String,
    pub train_id: i64,
    pub start_station_id: i64,
    pub end_station_id: i64,
}

#[derive(Deserialize)]
pub struct UpdateJourney {
    pub start_time: Option<DateTime<Utc>>,
    pub end_time: Option<DateTime<Utc>>,
    pub start_station_id: Option<i64>,
    pub end_station_id: Option<i64>,
}

#[derive(Debug, Serialize, sqlx::FromRow)]
pub struct JourneyBetweenStations {
    pub journey_id: i64,
    pub train_id: Option<i64>,
    pub train_name: Option<String>,
    pub start_station_id: Option<i64>,
    pub start_schedule_id: Option<i64>,
    pub start_station: Option<String>,
    pub end_station_id: Option<i64>,
    pub end_schedule_id: Option<i64>,
    pub end_station: Option<String>,
    pub start_time: Option<chrono::DateTime<Utc>>,
    pub end_time: Option<chrono::DateTime<Utc>>,
    pub start_stop_number: Option<i32>,
    pub end_stop_number: Option<i32>,
    pub travel_time: Option<i64>,
}